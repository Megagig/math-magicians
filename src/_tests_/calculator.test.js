import React from 'react';
import Calculator from '../components/calculator';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';

describe('Calculator component', () => {
  test('renders calculator with correct button labels', () => {
    render(<Calculator />);

    const button = [
      'AC',
      '+/-',
      '%',
      'รท',
      '7',
      '8',
      '9',
      'X',
      '4',
      '5',
      '6',
      '-',
      '1',
      '2',
      '3',
      '+',
      '0',
      '.',
      '=',
    ];

    for (const label of button) {
      const buttonEl = screen.queryAllByText(label);
      expect(buttonEl.length).toBeGreaterThanOrEqual(0);
    }
  });
});

describe('Calculator snapshot', () => {
  test('matches the snapshot', () => {
    const tree = renderer.create(<Calculator />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
