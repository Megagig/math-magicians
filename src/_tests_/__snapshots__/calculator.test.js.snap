// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator snapshot matches the snapshot 1`] = `
<div
  className="content-container"
>
  <div
    className="separation"
  >
    <h1
      className="title"
    >
      Let's do some math!
    </h1>
  </div>
  <input
    className="show result"
    readOnly={true}
    type="text"
    value="0"
  />
  <div
    className="btn-container"
  >
    <button
      className="btn button grey-button"
      onClick={[Function]}
      type="button"
    >
      AC
    </button>
    <button
      className="btn button grey-button"
      onClick={[Function]}
      type="button"
    >
      +/-
    </button>
    <button
      className="btn button grey-button"
      onClick={[Function]}
      type="button"
    >
      %
    </button>
    <button
      className="btn button operators-buttons"
      onClick={[Function]}
      type="button"
    >
      รท
    </button>
  </div>
  <div
    className="btn-container"
  >
    <button
      className="btn button grey-button"
      onClick={[Function]}
      type="button"
    >
      7
    </button>
    <button
      className="btn button grey-button"
      onClick={[Function]}
      type="button"
    >
      8
    </button>
    <button
      className="btn button grey-button"
      onClick={[Function]}
      type="button"
    >
      9
    </button>
    <button
      className="btn button operators-buttons"
      onClick={[Function]}
      type="button"
    >
      x
    </button>
  </div>
  <div
    className="btn-container"
  >
    <button
      className="btn button grey-button"
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      className="btn button grey-button"
      onClick={[Function]}
      type="button"
    >
      5
    </button>
    <button
      className="btn button grey-button"
      onClick={[Function]}
      type="button"
    >
      6
    </button>
    <button
      className="btn button operators-buttons"
      onClick={[Function]}
      type="button"
    >
      -
    </button>
  </div>
  <div
    className="btn-container"
  >
    <button
      className="btn button grey-button"
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      className="btn button grey-button"
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      className="btn button grey-button"
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      className="btn button operators-buttons"
      onClick={[Function]}
      type="button"
    >
      +
    </button>
  </div>
  <div
    className="btn-container"
  >
    <button
      className="btn button zero grey-button"
      onClick={[Function]}
      type="button"
    >
      0
    </button>
    <button
      className="btn button grey-button"
      onClick={[Function]}
      type="button"
    >
      .
    </button>
    <button
      className="btn button operators-buttons"
      onClick={[Function]}
      type="button"
    >
      =
    </button>
  </div>
</div>
`;
